### PR TITLE
Break undo if data snapshot is not changed.

### PR DESCRIPTION
**Pull request by ujjboy**
### Ⅰ. Describe what this PR did

- Add DataCompareUtils
- Add some test case.
- Do not execute `undo` if the before data snapshot is equivalent to the after data snapshot.

### Ⅱ. Does this pull request fix one issue?

Part of #1049

### CODE REVIEW

File: common/src/main/java/io/seata/common/util/StringUtils.java [hunk 1]
```diff
@@ -196,4 +197,32 @@ public static String toString(Object obj){
...
     }
+
+    /**
+     * Equals boolean.
+     *
+     * @param a the a
+     * @param b the b
+     * @return boolean
+     */
+    public static boolean equals(String a, String b) {
```
Comment (xingfudeshi) [inline common/src/main/java/io/seata/common/util/StringUtils.java]: What if the string is a blank character?should it be considered?

Comment (xingfudeshi) [reply common/src/main/java/io/seata/common/util/StringUtils.java]: like a="    "

Comment (ujjboy) [reply common/src/main/java/io/seata/common/util/StringUtils.java]: Do you mean  " "  and "" ?   I think there are not equals, maybe add new method `equalsIngoreBlank`

Comment (xingfudeshi) [reply common/src/main/java/io/seata/common/util/StringUtils.java]: yes

Comment (xingfudeshi) [reply common/src/main/java/io/seata/common/util/StringUtils.java]: > Do you mean " " and "" ? I think there are not equals, maybe add new method `equalsIngoreBlank`

can you add it in this PR,please?

Comment (ujjboy) [reply common/src/main/java/io/seata/common/util/StringUtils.java]: I will add it when we need it.

Comment (xingfudeshi) [reply common/src/main/java/io/seata/common/util/StringUtils.java]: > I will add it when we need it.

sure

**Pull request merged**